mkosi/config.py

```diff
@@ -936,7 +946,13 @@ def cache_manifest(self) -> dict[str, Any]:
 
     def to_json(self, *, indent: Optional[int] = 4, sort_keys: bool = True) -> str:
         """Dump MKosiConfig as JSON string."""
-        return json.dumps(dataclasses.asdict(self), cls=MkosiJsonEncoder, indent=indent, sort_keys=sort_keys)
+        def key_transformer(k: str) -> str:
+            if (s := SETTINGS_LOOKUP_BY_DEST.get(k)) is not None:
+                return s.name
+            return "".join(p.capitalize() for p in k.split("_"))
```

Comment:
Do we need these fallbacks at all for `MkosiConfig`? I think everything in there has an associated setting no?

Reply:
Yes, `preset` is an attribute of `MkosiConfig`, but there is no accompanying `MkosiConfigSetting`.